db/disk/submission.go

```
@@ -294,3 +294,17 @@ func (this *backend) getMostRecentSubmissionID(assignment *model.Assignment, ema
 
     return dirents[len(dirents) - 1].Name(), nil;
 }
+
+func (this *backend) RemoveSubmission(assignment *model.Assignment, email string, shortSubmissionID string) error {
+    submissionDir := this.getSubmissionDirFromAssignment(assignment, email, shortSubmissionID);
+    if (!util.PathExists(submissionDir)) {
+        return fmt.Errorf("No submission found");
+    }
+
+    err := os.RemoveAll(submissionDir)
+    if err != nil {
+        return fmt.Errorf("Error removing submission")
```

NEVER* ignore an error.

*99.9% of the time.